pyclustree/_clustree.py

```diff
@@ -99,6 +119,37 @@ def clustree(
         scatter_reference is None or scatter_reference in adata.obsm
     ), "The provided scatter reference is not valid. It should be present in adata.obsm."
 
+    if scatter_reference is not None and score_clustering is not None:
+        raise ValueError("Cluster scoring is not supported for scatter plotting.")
+
+    # Generate score array
+    score_array: Optional[list[Optional[float]]] = None
+
+    if score_clustering is not None:
+        score_array = [None] * len(cluster_keys)
+        for index, cluster_key in enumerate(cluster_keys):
+            # Assign basis for scoring
+            basis: Optional[NDArray] = None
+
+            if score_basis == "X":
+                basis = adata.X.copy()
+            elif score_basis == "pca":
+                basis = adata.obsm["X_pca"].copy()
+            elif score_basis == "raw":
+                basis = adata.raw.X.copy()
+
+            assert basis is not None, f"Can't score clustering on basis '{score_basis}'"
+
+            # Assign scoring function
+            if score_clustering == "calinski_harabasz":
+                score_array[index] = float(calinski_harabasz_score(X=basis, labels=adata.obs[cluster_key]))
+            elif score_clustering == "davies_bouldin":
+                score_array[index] = float(davies_bouldin_score(X=basis, labels=adata.obs[cluster_key]))
+            elif score_clustering == "silhouette":
+                score_array[index] = float(silhouette_score(X=basis, labels=adata.obs[cluster_key]))
+            else:
+                raise ValueError(f"Score '{score_clustering}' not a valid scoring method")
```

Comment:
Consider using dependency injection to allow users to provide their own scoring functions here while also parsing the strings. Importing these function should however also be conditional to keep import times short.